pkg/specgen/specgen.go

```diff
@@ -239,6 +239,9 @@ type ContainerStorageConfig struct {
 	// Devices are devices that will be added to the container.
 	// Optional.
 	Devices []spec.LinuxDevice `json:"devices,omitempty"`
+	// DeviceCGroupRule are device cgroup rules that allow containers
+	// to use additional types of devices.
+	DeviceCGroupRule []string `json:"device_cgroup_rule,omitempty"`
```

Comment:
Can we do the parsing on the frontend, so we can pass a struct here (including the type, path, and access permissions)?

Reply:
That would require all users of the API to do the processing. I believe this is something that Docker supports as well.
WDYT @jwhonce @baude

Reply:
Generally speaking, we try to parse in the frontend to the greatest extent possible, so the server does not have to do much validation of the contents of specgen.